### PR TITLE
Storybook: Deploy on PR, master and release

### PR DESCRIPTION
**Pull request by tskarhed**
Opened a new PR because of messy lockfile issues.
#22774
Closes #22006

### CODE REVIEW

**Review by aknuds1 — APPROVED** [review]
LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -411,6 +411,40 @@ jobs:
...
+            gcloud auth activate-service-account --key-file=/tmp/gcpkey.json
+            yarn install --frozen-lockfile --no-progress
+            yarn storybook:build
+            if [[ $CIRCLE_BRANCH == "chore/test-release-pipeline" ]]; then
+              # We're testing the release pipeline
+              echo Testing release
+            elif [[ $CIRCLE_BRANCH == "master" ]]; then
+              gsutil -m rsync -d -r ./packages/grafana-ui/dist/storybook gs://grafana-storybook/canary
+            elif [[ -n $CIRCLE_TAG ]]; then
+              gsutil -m rsync -d -r ./packages/grafana-ui/dist/storybook gs://grafana-storybook/$CIRCLE_TAG
```
Comment (dprokop) [inline .circleci/config.yml]: Minor: can we also put the release build into `latest` directory?

**Review by dprokop — APPROVED** [review]
Great, one comment to address and let's do it!

File: .circleci/config.yml (hunk 1 above)
Comment (tskarhed) [reply .circleci/config.yml]: Sounds like a good plan!

**Pull request merged**